src/lib.rs

```diff
@@ -44,10 +44,311 @@
+                ));
+            }
+            let hlp = File {
+                base_address: u64::from_str_radix(
+                    filevec[0].clone().drain(2..).collect::<String>().as_str(),
+                    16,
+                )
+                .unwrap(),
+                end: u64::from_str_radix(
+                    filevec[1].clone().drain(2..).collect::<String>().as_str(),
+                    16,
+                )
+                .unwrap(),
+                file_offset: u64::from_str_radix(
+                    filevec[3].clone().drain(2..).collect::<String>().as_str(),
+                    16,
+                )
+                .unwrap(),
+                name: match filevec.len() {
+                    0..=4 => "No_file".to_string(),
+                    _ => filevec[4].clone().to_string(),
+                },
+            };
+            some.push(hlp.clone());
+        }
+
+        Ok(MappedFiles {
+            fcount: some.len() as i32,
+            page_size: 4096,
+            files: some,
+        })
+    }
+
+    /// Method determines which file contains the address
+    ///
+    /// # Arguments
+    ///
+    /// * 'addr' - given address
+    pub fn find(&self, addr: u64) -> error::Result<File> {
+        let result = self
+            .files
+            .iter()
+            .find(|&x| (x.base_address < addr as u64) && (x.end > addr as u64));
+
+        if let Some(x) = result {
+            return Ok(x.clone());
+        } else {
+            return Err(error::Error::MF(
+                format!("Cannot find file with address {}", addr).to_string(),
+            ));
+        }
+    }
+}
+
+/// 'ModuleInfo' enum represents the name of the module or contains information about the module.
+#[derive(Clone)]
+pub enum ModuleInfo {
+    /// Module name
+    Name(String),
+    /// Module file
+    File(File),
+}
+
+/// `StacktraceEntry` struct represents the information about one line of the stacktrace.
+pub struct StacktraceEntry {
+    /// Function address
+    pub address: u64,
+    /// Information about the module
+    pub module: ModuleInfo,
+    /// Debug information
+    pub debug: String,
+}
+
+impl fmt::Display for StacktraceEntry {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "Address: 0x{:x}, ModuleInfo: {}, DebugInfo: {}",
+            self.address,
+            match self.module.clone() {
+                ModuleInfo::Name(x) => x,
+                ModuleInfo::File(x) => x.to_string(),
+            },
+            self.debug
+        )
+    }
+}
+
+/// Fucntion gets the stacktrace as a string and converts it into vector of 'StacktraceEntry' structs
+///
+/// # Arguments
+///
+/// * 'trace' - stacktrace from gdb
+///
+/// # Return value
+///
+/// The return value is a vector of  'StacktraceEntry' structs
+pub fn trace_from_gdb(trace: &String) -> error::Result<Vec<StacktraceEntry>> {
```

Comment:
Maybe struct Stacktrace(Vec<StacktraceEntry>) and create an implementation for this struct  impl Stactrace { from_gdb(trace:  &str)}}
```suggestion
struct  impl Stactrace { 
      pub fn from_gdb(trace:  &str) -> error::Result<Vec<StacktraceEntry>> {
      }
}
```